src/database/utilities/createRecord.js

```diff
@@ -359,7 +359,7 @@ const createTransactionBatch = (database, transactionItem, itemBatch) => {
     sellPrice,
     donor,
     transaction: transactionItem.transaction,
-    sortIndex: transactionItem.transaction ? transactionItem.transaction.numberOfBatches : 0,
+    sortIndex: transactionItem.transaction ? transactionItem.transaction.numberOfBatches + 1 : 1,
```

Comment:
Nice :) 

We have this [optional chaining](
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) syntax available, now - brand new!!!
It is a stage 4 proposal, so safe to use: https://github.com/tc39/proposal-optional-chaining

So this can become `(transactionItem?.transaction?.numberOfBatches || 0) + 1 || 1`

if you use [short circuit evaluation](https://codeburst.io/javascript-short-circuit-conditionals-bbc13ac3e9eb)

I think this is a lot more concise and nicer, what do you think?

Reply:
Thanks, this is great. Getting used to this syntax can take time at the beginning (hard to read to people that is not used to this maybe as well?) but it is a great to avoid deeply-nested structures. Thumbs up to this!